.github/workflows/changelog.yml

```diff
@@ -0,0 +1,43 @@
+name: Update CHANGELOG.md
+
+on:
+  pull_request_target:
+    branches:
+      - changelog-ci
+    types:
+      - closed
+  # For manually triggering this workflow
+  workflow_dispatch:
+    inputs:
+      number:
+        description: PR number
+        required: true
+env:
+  PR_TITLE: ${{ github.event.pull_request.title }} # this is returning empty
+  PR_NUMBER: ${{ github.event.number }}
+
+jobs:
+  update_changelog:
+    runs-on: ubuntu-latest
+    if: github.event.pull_request.merged == true
```

Comment:
Should also check the repository name here, so that it doesn't accidentally run on people's forks.